src/software/geom/spline.cpp

```diff
@@ -0,0 +1,55 @@
+#include "software/geom/spline.h"
+
+Spline::Spline(const std::vector<Point>& points)
+{
+    initLinearSegments(points);
+}
+
+Spline::Spline(const std::initializer_list<Point>& points)
+{
+    initLinearSegments(points);
+}
+
+Point Spline::calculateValue(double val) const
+{
+    if (val < 0)
+    {
+        return start_point;
+    }
+
+    if (val >= segments.size())
+    {
+        return end_point;
+    }
+
+    double x_val = segments[(size_t)val].x.calculateValue(val);
+    double y_val = segments[(size_t)val].y.calculateValue(val);
+    return Point(x_val, y_val);
+}
+
+size_t Spline::size(void) const
+{
+    return segments.size();
+}
+
+void Spline::initLinearSegments(const std::vector<Point>& points)
+{
+    if (points.size() == 1)
```

Comment:
Why can't we have a spline with one point?

Reply:
what would a one point spline look like?

Reply:
would would be the size of segments?

Reply:
Hmm, consider this from the perspective of an external user. It's entirely reasonable that you might have some spline that maps from `[0,1] -> {1,1}` so that ever input in the range maps to the same point. In general for these sorts of things we should make the external interface as close to the math as possible, then decide on an external implementation. 

As to your question about what the size of `segments` should wbe, a spline is defined by a series of `n` points, and `n-1` integers defining the order of the polynomials between the points. Storing it as a series of segments doesn't really make sense (assuming we want at least C0 continuity).

Reply:
done